openapi_python_client/parser/properties/__init__.py

```diff
@@ -167,18 +169,43 @@ def __attrs_post_init__(self) -> None:
             self, "has_properties_without_templates", any(prop.template is None for prop in self.inner_properties)
         )
 
-    def get_type_string(self, no_optional: bool = False) -> str:
-        """ Get a string representation of type that should be used when declaring this property """
-        inner_types = [p.get_type_string(no_optional=True) for p in self.inner_properties]
-        inner_prop_string = ", ".join(inner_types)
-        type_string = f"Union[{inner_prop_string}]"
-        if no_optional:
-            return type_string
-        if not self.required:
-            type_string = f"Union[Unset, {inner_prop_string}]"
-        if self.nullable:
-            type_string = f"Optional[{type_string}]"
-        return type_string
+    def _get_inner_type_strings(self, json: bool = False) -> List[str]:
+        inner_types = [p.get_type_string(no_optional=True, json=json) for p in self.inner_properties]
+        unique_inner_types = list(dict.fromkeys(inner_types))
+        return unique_inner_types
+
+    def _get_type_string_from_inner_type_strings(self, inner_types: List[str]) -> str:
+        if len(inner_types) == 1:
+            return inner_types[0]
+        else:
+            return f"Union[{', '.join(inner_types)}]"
```

Comment:
```suggestion
            return f"Union[{', '.join(sorted(inner_types))}]"
```
I think this will fix the determinism issue? While also maintaining the performance and reduced code of using sets.